lodash/lodash.d.ts

```diff
@@ -263,12 +263,12 @@ declare module _ {
         * @param array Array to compact.
         * @return (Array) Returns a new array of filtered values.
         **/
-        compact<T>(array: Array<T>): T[];
+        compact<T>(array?: Array<T>): T[];
```

Comment:
It might be more of a discussion point than anything else, but my reasoning for introducing this is because `_.compact()` or `_.compact(void(0))` yields `[]`. They might label it as a required argument in the documentation, but in practice, it's not actually required. It's because of this that I can do.

```ts
var foo: string[];
if (someCondition) {
  foo = ['bar', 'baz', 'qux'];
}
if (someOtherCondition) {
  foo = ['corge', 'garpley', 'thud'];
}
foo = _.compact(foo);
```

There's no reason that shouldn't compile is my point. It's totally valid code and because lodash functions don't throw errors, it fails gracefully and yields an empty array.

What are your thoughts on the matter?